src/app/child-dev-project/children/children.service.ts

```diff
@@ -197,13 +203,38 @@ export class ChildrenService {
         },
       },
     };
+
+    // TODO: remove these and use general note_by_relatedEntities instead?
     // creating a by_... view for each of the following properties
     ["children", "schools", "authors"].forEach(
       (prop) =>
         (designDoc.views[`by_${prop}`] = this.createNotesByFunction(prop))
     );
 
-    return this.dbIndexing.createIndex(designDoc);
+    await this.dbIndexing.createIndex(designDoc);
+
+    const newDesignDoc = {
+      _id: "_design/notes_related_index",
+      views: {
+        note_by_relatedEntities: {
+          map: `(doc) => {
+            if (!doc._id.startsWith("${Note.ENTITY_TYPE}") && !doc._id.startsWith("${EventNote.ENTITY_TYPE}")) return;
+            if (!Array.isArray(doc.relatedEntities)) return;
+
+            var d = new Date(doc.date || null);
+            var dateString = d.getFullYear() + "-" + String(d.getMonth()+1).padStart(2, "0") + "-" + String(d.getDate()).padStart(2, "0")
+
+            doc.relatedEntities.forEach((relatedEntity) => {
+              emit(relatedEntity + "_" + dateString);
+            });
+          }`,
+          // TODO: alternatively, we could build an index that combines all existing indexed properties for backward compatibility
```

Comment:
This, or we need to have a migration of existing data (which should not be too difficult though).

Reply:
discussed and postponed (see https://github.com/Aam-Digital/ndb-core/issues/1501#issuecomment-1308549111 )